group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -0,0 +1,1113 @@
+            "foo",
+            true
+        );
+
+        // Verify that the request is rejected with the correct exception.
+        assertThrows(UnknownMemberIdException.class, () -> context.commitOffset(
+            new OffsetCommitRequestData()
+                .setGroupId("foo")
+                .setMemberId("member")
+                .setGenerationIdOrMemberEpoch(10)
+                .setTopics(Collections.singletonList(
+                    new OffsetCommitRequestData.OffsetCommitRequestTopic()
+                        .setName("bar")
+                        .setPartitions(Collections.singletonList(
+                            new OffsetCommitRequestData.OffsetCommitRequestPartition()
+                                .setPartitionIndex(0)
+                                .setCommittedOffset(100L)
+                        ))
+                ))
+            )
+        );
+    }
+
+    @Test
+    public void testGenericGroupOffsetCommitWithIllegalGeneration() {
+        OffsetMetadataManagerTestContext context = new OffsetMetadataManagerTestContext.Builder().build();
+
+        // Create an empty group.
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup(
+            "foo",
+            true
+        );
+
+        // Add member.
+        group.add(new GenericGroupMember(
+            "member",
+            Optional.of("new-instance-id"),
+            "client-id",
+            "host",
+            5000,
+            5000,
+            "consumer",
+            new JoinGroupRequestData.JoinGroupRequestProtocolCollection(
+                Collections.singletonList(new JoinGroupRequestData.JoinGroupRequestProtocol()
+                    .setName("range")
+                    .setMetadata(new byte[0])
+                ).iterator()
+            )
+        ));
+
+        // Transition to next generation.
+        group.transitionTo(GenericGroupState.PREPARING_REBALANCE);
+        group.initNextGeneration();
+        assertEquals(1, group.generationId());
+
+        // Verify that the request is rejected with the correct exception.
+        assertThrows(IllegalGenerationException.class, () -> context.commitOffset(
+            new OffsetCommitRequestData()
+                .setGroupId("foo")
+                .setMemberId("member")
+                .setGenerationIdOrMemberEpoch(10)
+                .setTopics(Collections.singletonList(
+                    new OffsetCommitRequestData.OffsetCommitRequestTopic()
+                        .setName("bar")
+                        .setPartitions(Collections.singletonList(
+                            new OffsetCommitRequestData.OffsetCommitRequestPartition()
+                                .setPartitionIndex(0)
+                                .setCommittedOffset(100L)
+                        ))
+                ))
+            )
+        );
+    }
+
+    @Test
+    public void testGenericGroupOffsetCommitWithUnknownInstanceId() {
```

Comment:
This test seems to be the same as  testGenericGroupOffsetCommitWithUnknownMemberId() but we include the instance id. Could we maybe create the member but have the instance id not exist?

Reply:
Good catch. I should have created a member here without the static id. Otherwise, this test is exactly the same as testGenericGroupOffsetCommitWithUnknownMemberId as you pointed out.

The expected error is still correct though. We return UnknownMemberIdException when the instance id does not exist.

Reply:
I figured the error was the same, but just wanted to confirm we were testing the correct path in the code.